torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -2222,6 +2222,40 @@ at::Tensor& AtenXlaType::neg_(at::Tensor& self) {
   return self;
 }
 
+at::Tensor AtenXlaType::nll_loss2d_backward(
+    const at::Tensor& grad_output, const at::Tensor& self,
+    const at::Tensor& target, const at::Tensor& weight, int64_t reduction,
+    int64_t ignore_index, const at::Tensor& total_weight) {
+  XLA_FN_COUNTER("xla::");
+  XLATensor self_tensor = bridge::GetXlaTensor(self);
+  XLATensor weight_tensor =
+      bridge::GetOrCreateXlaTensor(weight, self_tensor.GetDevice());
+  XLATensor total_weight_tensor;
+  if (weight.defined()) {
+    total_weight_tensor =
+        bridge::GetOrCreateXlaTensor(total_weight, self_tensor.GetDevice());
+  }
+  return bridge::AtenFromXlaTensor(XLATensor::nll_loss2d_backward(
+      bridge::GetXlaTensor(grad_output), self_tensor,
+      bridge::GetXlaTensor(target), weight_tensor, reduction, ignore_index,
+      total_weight_tensor));
+}
+
+std::tuple<at::Tensor, at::Tensor> AtenXlaType::nll_loss2d_forward(
+    const at::Tensor& self, const at::Tensor& target, const at::Tensor& weight,
+    int64_t reduction, int64_t ignore_index) {
+  XLA_FN_COUNTER("xla::");
+  XLATensor self_tensor = bridge::GetXlaTensor(self);
+  XLATensor total_weight = XLATensor::Create(
```

Comment:
Maybe here calling `full()` would be a bit nicer?

https://github.com/pytorch/xla/blob/7fc9c3a72b96aec4e75bbe81d7d3f7272c05509a/torch_xla/csrc/tensor_methods.cpp#L1170